### PR TITLE
Refactor core interface

### LINKED ISSUE #40: Interface of NodeView
`NodeView` supports accessing the edge ids that contain a node in three different ways:
```python
H = some_hypergraph()
H.nodes[1] == H.nodes(1) == H.members(1)
# -> True
```
I find this unpythonic. I'd much rather there was a single way of doing things if at all possible. If there was exactly one way of accessing the edges that contain a node, *which would you prefer?* If anyone can argue that we should support two (or more) ways of doing the same thing, I'm all ears.

Related to #20.

### PR DESCRIPTION
**Pull request by leotrs**
This one is pretty big so I suggest reviewing each commit in turn instead of the whole thing at once. This basically implements the behavior discussed here #40. In particular, there is (hopefully, I think), one way of accessing each thing. Among other things:

* `H['attr']` accesses hypergraph attributes.
* `H.nodes()` and `H.edges()` are no longer supported
* `H.nodes[]` and `H.edges[]` now access attributes
* `H.nodes.membership` is now the same as `H.nodes.members`. (I decided to keep members for simplicity and symmetry with `H.edges`)

There are a few kinks to work out, but I thought this PR was already big enough.

**As you review, please let me know if you think of any other test I should add to this PR.**

Also: there's a lot of noise in this PR because I decided to change `*_attrs` to `*_attr` just to be uniform with the variable names in the main `Hypergraph` class.

Closes #40

### CODE REVIEW

**Comment by nwlandry** [comment]
I simplified the view classes and removed the data views.

File: xgi/classes/reportviews.py [hunk 1]
```diff
@@ -171,104 +148,38 @@ def __getitem__(self, n):
...
-                f"{type(self).__name__} does not support slicing, "
-                f"try list(H.nodes)[{n.start}:{n.stop}:{n.step}]"
-            )
-        elif n not in self._nodes:
-            raise XGIError(f"The node {n} is not in the hypergraph")
-
-        return self._node_attr[n]
+        try:
+            return self._id_attrs[id]
+        except:
```
Comment (leotrs) [inline xgi/classes/reportviews.py:153]: Comment: it's generally a bad idea to catch every single Exception using a naked `except`. It's always recommended to specify the exception type. In this case it'd be `except KeyError` (I believe).

File: tests/classes/test_hypergraph.py [hunk 2]
```diff
@@ -155,3 +155,12 @@ def test_isolates(edgelist1):
     assert H.isolates() == {4}
     H.remove_isolates()
     assert 4 not in H
+
+
+def test_add_node_atr(edgelist1):
```
Comment (maximelucas) [inline tests/classes/test_hypergraph.py]: atr -> attr in function name?

**Review by maximelucas — APPROVED** [review]

File: xgi/classes/hypergraph.py [hunk 3]
```diff
@@ -162,19 +155,9 @@ def __len__(self):
...
-        -------
-        list
-           A list of the edges of which the specified node is a part.
-
-        Notes
-        -----
-        H[n] is the same as H.nodes[n]
-        """
-        return self._node[n]
+    def __getitem__(self, attr):
```
Comment (maximelucas) [inline xgi/classes/hypergraph.py:158]: Do we have a test for this new behaviour?

File: tests/classes/test_hypergraph.py [hunk 4]
```diff
@@ -173,3 +173,12 @@ def test_hypergraph_attr(edgelist1):
         H['color']
     H['color'] = 'red'
     assert H['color'] == 'red'
+
+
+def test_members(edgelist1):
```
Comment (maximelucas) [inline tests/classes/test_hypergraph.py:179]: Add a line to test when the node does not exist, and when it's not a member of any edge?
assert H.nodes.membership(0) == ?

H.add_node(0)
assert H.nodes.membership((0) == ?

Comment (nwlandry) [reply tests/classes/test_hypergraph.py:179]: Fixed. Now it returns an XGIError.
